MediaBrowser.Providers/MediaInfo/MediaInfoResolver.cs

```diff
@@ -0,0 +1,114 @@
+using System;
+using System.Globalization;
+using System.Linq;
+using Emby.Naming.Common;
+using MediaBrowser.Controller.MediaEncoding;
+using MediaBrowser.Model.Entities;
+using MediaBrowser.Model.Globalization;
+
+namespace MediaBrowser.Providers.MediaInfo
+{
+    /// <summary>
+    /// Resolves external audios for videos.
+    /// </summary>
+    public class MediaInfoResolver
+    {
+        /// <summary>
+        /// The <see cref="ILocalizationManager"/> instance.
+        /// </summary>
+        protected ILocalizationManager _localizationManager;
+
+        /// <summary>
+        /// The <see cref="IMediaEncoder"/> instance.
+        /// </summary>
+        protected IMediaEncoder _mediaEncoder;
+
+        /// <summary>
+        /// The <see cref="NamingOptions"/> instance.
+        /// </summary>
+        protected NamingOptions _namingOptions;
+
+        /// <summary>
+        /// The <see cref="CompareOptions"/> instance.
+        /// </summary>
+        protected const CompareOptions CompareOptions = System.Globalization.CompareOptions.IgnoreCase | System.Globalization.CompareOptions.IgnoreNonSpace | System.Globalization.CompareOptions.IgnoreSymbols;
+
+        /// <summary>
+        /// The <see cref="CompareInfo"/> instance.
+        /// </summary>
+        protected readonly CompareInfo _compareInfo = CultureInfo.InvariantCulture.CompareInfo;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MediaInfoResolver"/> class.
+        /// </summary>
+        /// <param name="localizationManager">The localization manager.</param>
+        /// <param name="mediaEncoder">The media encoder.</param>
+        /// <param name="namingOptions">The naming options.</param>
+        public MediaInfoResolver(
+            ILocalizationManager localizationManager,
+            IMediaEncoder mediaEncoder,
+            NamingOptions namingOptions)
+        {
+            _localizationManager = localizationManager;
+            _mediaEncoder = mediaEncoder;
+            _namingOptions = namingOptions;
+        }
+
+        /// <summary>
+        /// Detects the language of a specific media stream.
+        /// </summary>
+        /// <param name="mediaStream">The media stream.</param>
+        /// <param name="fileNameWithoutExtension">The filename without extension.</param>
+        /// <param name="videoFileNameWithoutExtension">The video filename without extension.</param>
```

Comment:
This is only ever used to check length, and it should really be the `matchLength` from the `_compareInfo.IsPrefix` call when creating `cleanedFiles` in `SubtitleResolver.GetExternalSubtitleFiles`.

This test case verifies the prefix is matched without considering punctuation (a subtitle is found) but demonstrates the problem with using the video name length instead of the match length (`en` is ignored because the `.` after the title isn't at the index it expects due to the subtitle filename being shorter than the video file name).
```
[InlineData("MyVideo.en.srt", "srt", "eng", null, false, false)]
```